src/main/java/io/cdap/plugin/sap/OData2Client.java

```diff
@@ -0,0 +1,170 @@
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.List;
+import javax.annotation.Nullable;
+import javax.ws.rs.HttpMethod;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.MediaType;
+
+/**
+ * Provides handy methods to consume OData v2 services using Apache Olingo v2 as Client Library.
+ */
+public class OData2Client {
+
+  private static final String METADATA = "$metadata";
+  private static final String SEPARATOR = "/";
+  private static final String QUERY_SEPARATOR = "?";
+
+  private String rootUrl;
+  private String username;
+  private String password;
+
+  /**
+   * @param rootUrl  URL of the OData service. The URL must end with an external service name
+   *                 (e.g., http://eccsvrname:8000/sap/opu/odata/sap/zgw100_dd02l_so_srv/).
+   * @param username username for basic authentication.
+   * @param password password for basic authentication.
+   */
+  public OData2Client(String rootUrl, String username, String password) {
+    this.rootUrl = rootUrl;
+    this.username = username;
+    this.password = password;
+  }
+
+  /**
+   * Query {@link ODataFeed} using the specified entity set name and OData query.
+   *
+   * @param entitySetName entity set name.
+   * @param query         query such as "$top=2&$select=BuyerName&$filter=BuyerName eq 'TECUM'".
+   * @return {@link ODataFeed} for the specified entity set name.
+   * @throws ODataException if the specified entity set cannot be read.
+   */
+  public ODataFeed queryEntitySet(String entitySetName, @Nullable String query) {
+    Edm metadata = getMetadata();
+
+    try {
+      EdmEntitySet entitySet = metadata.getDefaultEntityContainer().getEntitySet(entitySetName);
+      String entitySetUrl = Strings.isNullOrEmpty(query) ?
+        rootUrl + SEPARATOR + entitySetName : rootUrl + SEPARATOR + entitySetName + QUERY_SEPARATOR + query;
+      HttpURLConnection connection = connect(entitySetUrl, username, password);
+      InputStream content = (InputStream) connection.getContent();
+
+      return EntityProvider.readFeed(MediaType.APPLICATION_XML, entitySet, content,
+                                     EntityProviderReadProperties.init().build());
+    } catch (IOException | EdmException | EntityProviderException e) {
+      throw new ODataException(String.format("Unable to read '%s' entity set.", entitySetName), e);
+    }
+  }
+
+  /**
+   * Get OData service metadata.
+   *
+   * @return OData service metadata.
+   * @throws ODataException if the metadata cannot be fetched.
+   */
+  public Edm getMetadata() {
+    String metadataUrl = rootUrl + SEPARATOR + METADATA;
+    HttpURLConnection connection = connect(metadataUrl, username, password);
+    try {
+      InputStream content = connection.getInputStream();
+      return EntityProvider.readMetadata(content, false);
```

Comment:
does this method close the InputStream?

Reply:
Changed to call `connection.disconnect();` in the `finally` block. This will close the content `InputStream`.